libcontainer/nsenter/nsexec.c

```diff
@@ -1120,54 +775,11 @@ void nsexec(void)
 		 *          init_linux.go to run.
 		 */
 	case STAGE_INIT:{
-			/*
-			 * We're inside the child now, having jumped from the
-			 * start_child() code after forking in the parent.
-			 */
-			enum sync_t s;
-
 			/* For debugging. */
 			current_stage = STAGE_INIT;
-
-			/* We're in a child and thus need to tell the parent if we die. */
-			syncfd = sync_grandchild_pipe[0];
-			if (close(sync_grandchild_pipe[1]) < 0)
-				bail("failed to close sync_grandchild_pipe[1] fd");
-
-			if (close(sync_child_pipe[0]) < 0)
-				bail("failed to close sync_child_pipe[0] fd");
-
-			/* For debugging. */
-			prctl(PR_SET_NAME, (unsigned long)"runc:[2:INIT]", 0, 0, 0);
```

Comment:
I'd keep this. It's way simpler the init stage now, but still things can go wrong (specially since we are doing big changes) and it is useful to have this IMHO.

Reply:
👍 make sense.